specs/networking/rpc-interface.md

```diff
@@ -0,0 +1,246 @@
+
+Clients SHOULD immediately disconnect from one another following the handshake above under the following conditions:
+
+1. If `network_id` belongs to a different chain, since the client definitionally cannot sync with this client.
+2. If the `latest_finalized_root` shared by the peer is not in the client's chain at the expected epoch. For example, if Peer 1 in the diagram below has `(root, epoch)` of `(A, 5)` and Peer 2 has `(B, 3)`, Peer 1 would disconnect because it knows that `B` is not the root in their chain at epoch 3:
+
+```
+              Root A
+
+              +---+
+              |xxx|  +----+ Epoch 5
+              +-+-+
+                ^
+                |
+              +-+-+
+              |   |  +----+ Epoch 4
+              +-+-+
+Root B          ^
+                |
++---+         +-+-+
+|xxx+<---+--->+   |  +----+ Epoch 3
++---+    |    +---+
+         |
+       +-+-+
+       |   |  +-----------+ Epoch 2
+       +-+-+
+         ^
+         |
+       +-+-+
+       |   |  +-----------+ Epoch 1
+       +---+
+```
+
+Once the handshake completes, the client with the higher `latest_finalized_epoch` or `best_slot` (if the clients have equal `latest_finalized_epoch`s) SHOULD send beacon block roots to its counterparty via `beacon_block_roots` (i.e., RPC method `10`).
+
+### Goodbye
+
+**Method ID:** `1`
+
+**Body:**
+
+```
+(
+    reason: uint64
+)
+```
+
+Client MAY send `goodbye` messages upon disconnection. The reason field MUST be one of the following values:
+
+- `1`: Client shut down.
+- `2`: Irrelevant network.
+- `3`: Irrelevant shard.
+
+### Provide Beacon Block Roots
+
+**Method ID:** `10`
+
+**Body:**
+
+```
+# BlockRootSlot
+(
+    block_root: HashTreeRoot
+    slot: uint64
+)
+
+(
+    roots: []BlockRootSlot
+)
+```
+
+Send a list of block roots and slots to the peer.
+
+### Beacon Block Headers
+
+**Method ID:** `11`
+
+**Request Body**
+
+```
+(
+    start_root: HashTreeRoot
+    start_slot: uint64
+    max_headers: uint64
+    skip_slots: uint64
+)
+```
+
+**Response Body:**
+
+```
+(
+    headers: []BlockHeader
+)
+```
+
+Requests beacon block headers from the peer starting from `(start_root, start_slot)`. The response MUST contain no more than `max_headers` headers. `skip_slots` defines the maximum number of slots to skip between blocks. For example, requesting blocks starting at slots `2` a `skip_slots` value of `2` would return the blocks at `[2, 4, 6, 8, 10]`. In cases where a slot is undefined for a given slot number, the closest previous block MUST be returned. For example, if slot `4` were undefined in the previous example, the returned array would contain `[2, 3, 6, 8, 10]`. If slot three were further undefined, the array would contain `[2, 6, 8, 10]` - i.e., duplicate blocks MUST be collapsed.
```

Comment:
`skip_slots` sounds off-by-one to me: skipping 0 blocks should result in a continuous sequence.

Reply:
I think this makes more sense from an API perspective. Will fix.